onesignal-admin.php

```diff
@@ -272,12 +272,12 @@ public static function onesignal_notif_on_post_html_view($post)
         $settings_send_notification_on_wp_editor_post = $onesignal_wp_settings['notification_on_post'];
 
         /* This is a scheduled post and the user checked "Send a notification on post publish/update". */
-        $post_metadata_was_send_notification_checked = (get_post_meta($post->ID, 'onesignal_send_notification') === true);
+        $post_metadata_was_send_notification_checked = (get_post_meta($post->ID, 'onesignal_send_notification', true) === '1');
 
         // We check the checkbox if: setting is enabled on Config page, post type is ONLY "post", and the post has not been published (new posts are status "auto-draft")
         $meta_box_checkbox_send_notification = ($settings_send_notification_on_wp_editor_post &&  // If setting is enabled
                                             $post->post_type === 'post' &&  // Post type must be type post for checkbox to be auto-checked
-                                            in_array($post->post_status, array('future', 'draft', 'auto-draft', 'pending'), true)) || // Post is scheduled, incomplete, being edited, or is awaiting publication
+                                            !in_array($post->post_status, array('publish', 'private', 'trash', 'inherit'), true)) || // Post is scheduled, incomplete, being edited, or is awaiting publication
```

Comment:
Do we need to check a parent post if it is `inherit`? Not sure what this status is exactly.

Reply:
Inherit: Used with a child post (such as Attachments and Revisions) to determine the actual status of the parent post. (inherit)